packages/devtools_app/integration_test/test_infra/run/_in_file_args.dart

```diff
@@ -0,0 +1,62 @@
+// Copyright 2023 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:io';
+
+enum InFileArgItems {
+  experimentsOn,
+  appPath,
+}
+
+/// Test arguments, defined inside the test file as a comment.
+class InFileArgs {
+  factory InFileArgs(String testFilePath) {
+    final content = File(testFilePath).readAsStringSync();
+    return InFileArgs.fromFileContent(content);
+  }
+
+  factory InFileArgs.fromFileContent(String fileContent) {
+    final values = _parseFileContent(fileContent);
+
+    for (final arg in values.keys) {
+      values.putIfAbsent(arg, () => null);
+    }
+
+    return InFileArgs.fromValues(values);
+  }
+
+  InFileArgs.fromValues(Map<InFileArgItems, dynamic> values)
+      : experimentsOn = values[InFileArgItems.experimentsOn] ?? false,
+        appPath = values[InFileArgItems.appPath] ??
+            'test/test_infra/fixtures/flutter_app';
+
+  /// If true, experiments will be enabled in the test.
+  final bool experimentsOn;
+
+  /// Path to the application to connect to.
+  final String appPath;
+}
+
+const _argPrefix = '// test-argument:';
```

Comment:
I think this should be more permissive even if we're going to enforce this format in code review. Changing this to a regex will allow for more flexibility for whitespace (which means less head scratching when someone's writing a test for the first time and wonders why `//test-argument:foo=bar` doesn't work) and will also simplify parsing by using groups.

Here's an example pattern that could work where group(1) is the matched key and group(2) is the matched value:

`\/\/\s*test-arguments:\s*(\w*)=(\w*)`

Reply:
Yes, I started with regex, but then found that code readability became much worse. This means people will scratch their heads maintaining this code. 
So, this is choice between two downsides. So I prefer to keep current option.

Reply:
Nothing a little documentation wouldn't fix... :-)

I still prefer the regex since it's less brittle and more permissive, but maybe @kenzieschmoll or @jacob314 will have an opinion.

Reply:
Let's save their time. I will switch to regex :)